src/main/java/com/zufar/icedlatte/review/api/ReviewCreator.java

```diff
@@ -0,0 +1,54 @@
+package com.zufar.icedlatte.review.api;
+
+import com.zufar.icedlatte.openapi.dto.NewReview;
+import com.zufar.icedlatte.openapi.dto.ReviewResponse;
+import com.zufar.icedlatte.product.exception.ProductNotFoundException;
+import com.zufar.icedlatte.product.repository.ProductInfoRepository;
+import com.zufar.icedlatte.review.converter.ReviewDtoConverter;
+import com.zufar.icedlatte.review.entity.Review;
+import com.zufar.icedlatte.review.exception.UnsupportedReviewFormatException;
+import com.zufar.icedlatte.review.repository.ReviewRepository;
+import com.zufar.icedlatte.security.api.SecurityPrincipalProvider;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.OffsetDateTime;
+import java.util.UUID;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ReviewCreator {
+
+    private final ReviewRepository reviewRepository;
+    private final ProductInfoRepository productInfoRepository;
+    private final ReviewDtoConverter reviewDtoConverter;
+    private final SecurityPrincipalProvider securityPrincipalProvider;
+
+    @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.READ_COMMITTED)
+    public ReviewResponse create(final UUID productId, final NewReview newReview) {
+        UUID userId = securityPrincipalProvider.getUserId();
+        var text = newReview.getText().trim();
+        if (text.isEmpty()) {
+            throw new UnsupportedReviewFormatException();
+        }
+        var product = productInfoRepository.findById(productId);
```

Comment:
I think you should create an additional method in the product class that contains "...repository.existsById(producrId)"

Reply:
Yes, I agree 👍 I think I'll leverage yours [`getProductEntityById`](https://github.com/Sunagatov/Iced-Latte/pull/246/commits/e611c7288c49c80bc05b255205e492372784b365#diff-6b53a46d0a231c949bcbe37bc56b1c557f1bc2120b295828b3db3eb8161de41dR41)

Reply:
Fixed